src/Firestore/Codec.elm

```diff
@@ -0,0 +1,227 @@
+module Firestore.Codec exposing
+    ( Codec, asEncoder, encode, asDecoder, decode
+    , Document, document, build, required, optional
+    , Field, bool, bytes, int, string, list, dict, null, maybe, timestamp, geopoint, reference
+    , map, andThen, succeed, fail, construct
+    )
+
+{-| Codec for Firestore
+
+@docs Codec, asEncoder, encode, asDecoder, decode
+
+
+# Constructors
+
+@docs Document, document, build, required, optional
+
+
+# Field
+
+@docs Field, bool, bytes, int, string, list, dict, null, maybe, timestamp, geopoint, reference
+
+
+# Advanced
+
+@docs map, andThen, succeed, fail, construct
+
+-}
+
+import Dict exposing (Dict)
+import Firestore.Decode as Decode exposing (Decoder)
+import Firestore.Encode as Encode exposing (Encoder)
+import Firestore.Types.Geopoint exposing (Geopoint)
+import Firestore.Types.Reference exposing (Reference)
+import Json.Decode as D
+import Json.Encode as E
+import Time exposing (Posix)
+
+
+{-| -}
+type Codec a
+    = Codec (Decoder a) (a -> Encoder)
+
+
+{-| -}
+asEncoder : Codec a -> (a -> Encoder)
+asEncoder (Codec _ out) =
+    out
+
+
+{-| -}
+asDecoder : Codec a -> Decoder a
+asDecoder (Codec out _) =
+    out
+
+
+{-| -}
+encode : Codec a -> a -> E.Value
+encode codec =
+    asEncoder codec
+        >> Encode.encode
+
+
+{-| -}
+decode : Codec a -> D.Decoder a
+decode =
+    asDecoder
+        >> Decode.decode
+
+
+
+--------------------------------------------------------------------------------
```

Comment:
Could you write section comment here instead of a separetor? 

I mean I would like here to be like as `-- Constructors`. Other lines after here are the same.